### ISSUE TITLE
[SPIKE]: Plan how to modernize Laboratory codebase

### ISSUE THREAD

**Issue by piyalbasu** [stellar/laboratory#523]
This is a pretty old codebase with some outdated conventions. How can we develop a strategy to modernize?

Things we would like to do:
- Implement Typescript
- Be able to add tests
- Create reusable/exportable const's
- Implement latest ES spec for older browsers: (optional chaining, etc)
- Streamline some flows so it's easier to understand
- Add more documentation so other engineers can easily jump in and make updates

**Comment by shanzzam** [comment]
Outcome of this spike:

- [ ]  High-level architecture document outlining phased work and overall strategy
- [ ] Phased plan made up of smaller work items 
- [ ] Pro/Con evaluation of what we gain by doing this.

**Comment by piyalbasu** [comment]
Confluence doc of my findings are here:
https://stellarorg.atlassian.net/wiki/spaces/PET/pages/362643469/Laboratory+Modernization

TLDR: The codebase isn't in that bad of shape, we just need to modernize it before we can start building anything new on top of it. We can take it one file at a time and add types, fix linting issues, convert old React API to new React API, etc etc. over a long period of time without much issue.

**Comment by shanzzam** [comment]
Plan review and is solid. Next step is to produce a roadmap for the work. Either we can set time aside to do the work internally or enlist the help of the community.
